src/rpucuda/cuda/weight_modifier_cuda.cu

```diff
@@ -462,6 +539,17 @@ void WeightModifierCuda<T>::apply(
     }
     break;
   }
+  case WeightModifierType::DiscretizePerChannel: {
+    if (wmpar.res > (T)0.0) {
+      kernelModifyWeightsDiscretizePerChannel<T><<<nblocks, nthreads, 0, s>>>(
+          size_, d_size_, wmpar.copy_last_column, new_weights, weights, wmpar.res, wmpar.sto_round,
+          wmpar.assumed_wmax, amax,
+          wmpar.sto_round ? context_->getRandomStates(nblocks * nthreads) : nullptr);
+      cudaFree(amax);
```

Comment:
I do the discretization and free the memory again that was used for the per-column amax.